src/common/class-syntax.k

```diff
@@ -30,22 +31,15 @@ syntax ClassMemberDec ::= FieldDec | ClassDec | MethodDec | InterfaceDec | SemiC
 syntax SemiColon ::= ";" [klabel('SemiColon)]
 
 //@ \subsection{ConstructorDeclarations.sdf}
-//todo optionals
-syntax ConstrHead   ::=  AnnoConstrModList TypeParams Type Id "(" Params ")" Throws [prefer, klabel('ConstrDecHead)]
-                       | AnnoConstrModList TypeParams Type Id "(" Params ")"    [klabel('ConstrDecHead)]
-                       | AnnoConstrModList Type Id "(" Params ")" Throws   [klabel('ConstrDecHead)]
-                       | AnnoConstrModList Type Id "(" Params ")"    [klabel('ConstrDecHead)]
+syntax ConstrHead   ::=  AnnoConstrModList OptionalTypeParams Type
+                        Id "(" Params ")" OptionalThrows [klabel('ConstrDecHead)]
```

Comment:
Not particularly. I would recommend you look at the C semantics though. It has relatively decent rules about where and how to break lines in rules.